5.0 Draft/IDD/IDDs Orchestration System/eu.arrowhead.orchestration-pull-http-json.yml

```diff
@@ -0,0 +1,303 @@
+      type: object
+      example: {"interfaceName1":"token1"}
+
+    interface:
+      description: A string that describes the interface in Protocol-SecurityType-MimeType format. SecurityType can be either SECURE or 
+        INSECURE. 
+      type: string
+      pattern: ^([a-zA-Z0-9_-]+)-(SECURE|INSECURE)-([a-zA-Z0-9_-]+)$
+      example: HTTP-SECURE-JSON
+    
+    Metadata:    
+      description: |
+        Additional details of relevance when consuming a service. What metadata is made available
+        depends on the type of the service holding the metadata. Possible examples of metadata are
+        HTTP base paths, message size restrictions and caching directives.
+      type: object
+      example: {"basePath":"v2"}
+
+    PreferredProvider:
+      type: object
+      description: information of a preferred provider to consume services from.
+      properties:
+        provider-cloud:
+          $ref: '#/components/schemas/ProviderCloud'
+        provider:
+          $ref: '#/components/schemas/ProviderSystem'
+
+    ProviderSystem:
+      type: object
+      description: information describing a provider system within a local cloud. 
+      properties:
+        system-name:
+          type: string
+        address:
+          $ref: '#/components/schemas/Address'
+    
+    ProviderCloud:
+      type: object
+      description: an object describing the cloud in which a provider is located
+      properties:
+        operator:
+          type: string
+          description: Name of the operator company
+        name: 
+          type: string
+          description: name of the cloud 
+
+    RequesterSystem:
+      type: object
+      description: Information of the system requesting orchestration information. 
+      properties:
+        system-name: 
+          type: string
+          description: Short human readable system name.
+        systemID:
+          type: string
+          description: unique identifier of requester system
+        address: 
+          $ref: '#/components/schemas/Address'
+        auth-info:
+          type: string
+          description: Authentication info, i.e, CERTIFICATE, TOKEN, etc. 
+        interface: 
+          type: array
+          items:
+            $ref: '#/components/schemas/Interface'
+
+    RequestedService:
+      type: object
+      description: |
+        Optional information of the required service. This object contains the service requirements of the requester
+        system. 
+      properties: 
+        service-definition:
+          type: string
+          description: Service definition 
+        interfaces:
+          type: array
+          items:
+            $ref: '#/components/schemas/Interface'
+        security: 
+          type: array
+          items:
+            $ref: '#/components/schemas/SecurityType'
+        metadata:
+          $ref: '#/components/schemas/Metadata'
+        version:
+          $ref: '#/components/schemas/VersionReq'
+
+    Service:
```

Comment:
Rather than copying and pasting types between specifications, you can move them out into separate files and refer to those files from all specifications that need them.

The `$ref` tag can take a relative filesystem path, like `"$ref": "../Shared Types/Service.yml`. In this case, it would contain everything hierarchically ordered below the `Service` label on this line.

Reply:
How would you say is the better way to go for it? Should we have at the IDD folder level some sort of "Resources" or "Data models" document which is referred by the IDDs? Should those "Shared Resources" be by System?

Reply:
Please try to keep the IDD:s as independent as possible. I think it is okay to duplicate information in order to keep the specifications atomic.